### PR TITLE
VPN-6050: Change checkboxes to toggles

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## Description

- Add `MZToggleRow` component (for toggles with labels, similar to [MZCheckBoxRow](https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/nebula/ui/components/MZCheckBoxRow.qml)
- Replaces checkboxes with toggles on 
  - Onboarding data slide
  - Onboarding privacy slide
  - Onboarding start slide (desktop)
  - Privacy feature settings
  - App preferences
  - Notification settings
- Add dividers to
  - Privacy feature list (onboarding and settings)
  - App preferences
  - Notification settings

## Reference

[VPN-6050: UX Improvement: Switch checkboxes to toggles](https://mozilla-hub.atlassian.net/browse/VPN-6050)

### CODE REVIEW

**Review by flodolo — APPROVED** [review]

File: nebula/ui/components/MZSettingsToggle.qml [hunk 1]
```diff
@@ -147,12 +139,12 @@ CheckBox {
     focusPolicy: Qt.StrongFocus
 
     Keys.onReleased: event => {
-        if (event.key === Qt.Key_Return)
+        if (event.key === Qt.Key_Return || event.key === Qt.Key_Space)
             uiPlaceholder.state = uiState.stateDefault;
     }
 
     Keys.onPressed: event => {
-        if (event.key === Qt.Key_Return || event.key === Qt.Key_Space)
```
Comment (lesleyjanenorton) [inline nebula/ui/components/MZSettingsToggle.qml:155]: do we have guidance from a11y that this is correct or do we get the Key_Space click for free somehow?

File: src/ui/screens/onboarding/OnboardingDataSlide.qml [hunk 2]
```diff
@@ -21,39 +21,49 @@ ColumnLayout {
...
+            text: MZI18n.OnboardingDataSlideToggleLabel
+            font.pixelSize: MZTheme.theme.fontSize
+            lineHeight: MZTheme.theme.labelLineHeight
+        }
 
-        onClicked: MZSettings.onboardingDataCollectionEnabled = !MZSettings.onboardingDataCollectionEnabled
+        MZSettingsToggle {
+            objectName: "dataCollectionToggle"
+
+            implicitHeight: 24
```
Comment (lesleyjanenorton) [inline src/ui/screens/onboarding/OnboardingDataSlide.qml:61]: Why not use the theme vars here?

File: nebula/ui/components/MZSettingsToggle.qml [hunk 3]
```diff
@@ -58,12 +56,6 @@ CheckBox {
         }
     ]
 
-    MZToolTip {
```
Comment (lesleyjanenorton) [inline nebula/ui/components/MZSettingsToggle.qml:61]: Just curious, why remove the tooltip?

File: src/ui/screens/onboarding/OnboardingStartSlideDesktop.qml [hunk 4]
```diff
@@ -34,40 +34,52 @@ ColumnLayout {
...
-                });
+            text: MZI18n.SettingsStartAtBootTitle
+            font.pixelSize: MZTheme.theme.fontSize
+            lineHeight: MZTheme.theme.labelLineHeight
+        }
+
+        MZSettingsToggle {
+            objectName: "startAtBootToggle"
+
+            implicitHeight: 24
```
Comment (lesleyjanenorton) [inline src/ui/screens/onboarding/OnboardingStartSlideDesktop.qml:64]: Same theme vars comment as before. Also, do we expect the sizing here to ever change? Could we specify this in MZSettingsToggle and then just overwrite it when necessary?

**Review by lesleyjanenorton — APPROVED** [review]
R+ . A few questions/comments but nothing blocking. Thanks @MattLichtenstein !

File: nebula/ui/components/MZSettingsToggle.qml (hunk 1 above)
Comment (MattLichtenstein) [reply nebula/ui/components/MZSettingsToggle.qml:155]: We get space for free. this change was merely to fix ui states

File: src/ui/screens/onboarding/OnboardingDataSlide.qml (hunk 2 above)
Comment (MattLichtenstein) [reply src/ui/screens/onboarding/OnboardingDataSlide.qml:61]: Oops, `implicitHeight` and `implicitWidth` are already set (using the theme vars) inside of `MZSettingsToggle` already. will remove!

File: nebula/ui/components/MZSettingsToggle.qml (hunk 3 above)
Comment (MattLichtenstein) [reply nebula/ui/components/MZSettingsToggle.qml:61]: All of our toggles have labels next to them, so the tooltips were unnecessary. similarly none of our checkboxes had tooltips

**Pull request merged**